net/dns/manager.go

```diff
@@ -454,17 +463,61 @@ func (s *dnsTCPSession) handleReads() {
 // servicing DNS requests sent down it.
 func (m *Manager) HandleTCPConn(conn net.Conn, srcAddr netaddr.IPPort) {
 	s := dnsTCPSession{
-		m:            m,
-		conn:         conn,
-		srcAddr:      srcAddr,
-		responses:    make(chan []byte),
-		readClosing:  make(chan struct{}),
+		m:           m,
+		conn:        conn,
+		srcAddr:     srcAddr,
+		responses:   make(chan []byte),
+		readClosing: make(chan struct{}),
 	}
 	s.ctx, s.closeCtx = context.WithCancel(context.Background())
 	go s.handleReads()
 	s.handleWrites()
 }
 
+// HandleTCPConn implements magicDNS over DNS-over-TLS, taking a connection and
+// servicing DNS requests sent down it.
+func (m *Manager) HandleDNSoverTLSConn(conn net.Conn, srcAddr netaddr.IPPort) {
```

Comment:
I don't understand how this will work to Android's satisfaction. The TLS cert will be invalid, and we have no way of generating a valid one. Does android just not care?...

Reply:
Is there an upstream bug we can track that says that android doesn't care?

Reply:
Yeah see the bug. There are two modes.